pennylane/labs/trotter_error/product_formulas/error.py

```diff
@@ -265,21 +328,26 @@ def _get_expval_state(commutator_lists, fragments, state: AbstractState, timeste
     """Returns the expectation value of ``state`` with respect to the operator obtained by substituting ``fragments`` into ``commutators``."""
 
     expectations = {}
+    logs = {}
     for commutators in commutator_lists:
         if len(commutators) == 0:
             continue
 
         order = len(commutators[0])
-        expectation = sum(
-            _compute_expectation(commutator, fragments, state) for commutator in commutators
-        )
+        expectation = 0
+        log = ConvergenceLog()
+        for commutator in commutators:
+            expectation += _compute_expectation(commutator, fragments, state)
+            log = _update_convergence_log(expectation, log)
```

Comment:
I am bit concerned with this. We might check it broadly to see what is the recommended strategy to handle such cases in PL.